cpp/include/cudf/detail/sizes_to_offsets_iterator.cuh

```diff
@@ -269,5 +273,55 @@ auto sizes_to_offsets(SizesIterator begin,
   return last_element.value(stream);
 }
 
+/**
+ * @brief Create an offsets column to be a child of a compound column
+ *
+ * This function sets the offsets values by executing scan over the sizes in the provided
+ * Iterator.
+ *
+ * The return also includes the total number of elements -- the last element value from the
+ * scan.
+ *
+ * @throw cudf::logic_error if the total size of the scan (last element) greater than maximum value
+ * of `size_type`
+ *
+ * @tparam Iterator Used as input to scan to set the offset values
+ * @param begin The beginning of the input sequence
+ * @param end The end of the input sequence
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ * @return Offsets column and total elements
+ */
+template <typename InputIterator>
+std::pair<std::unique_ptr<column>, size_type> make_offsets_child_column(
+  InputIterator begin,
+  InputIterator end,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr)
+{
+  auto count          = static_cast<size_type>(std::distance(begin, end));
+  auto offsets_column = make_numeric_column(
+    data_type{type_to_id<offset_type>()}, count + 1, mask_state::UNALLOCATED, stream, mr);
+  auto offsets_view = offsets_column->mutable_view();
+  auto d_offsets    = offsets_view.template data<offset_type>();
+
+  // The number of offsets is count+1 so to build the offsets from the sizes
+  // using exclusive-scan technically requires count+1 input values even though
+  // the final input value is never used.
+  // The input iterator is wrapped here to allow the last value to be safely read.
+  auto map_fn = [begin, count] __device__(size_type idx) -> size_type {
+    return idx < count ? static_cast<size_type>(begin[idx]) : size_type{0};
+  };
+  auto input_itr = cudf::detail::make_counting_transform_iterator(0, map_fn);
+  // Use the sizes-to-offsets iterator to compute the total number of elements
+  auto const total_elements = sizes_to_offsets(input_itr, input_itr + count + 1, d_offsets, stream);
+  CUDF_EXPECTS(
+    total_elements <= static_cast<decltype(total_elements)>(std::numeric_limits<size_type>::max()),
+    "Size of output exceeds column size limit");
+
+  offsets_column->set_null_count(0);
```

Comment:
Is `null_count == 0` implied by the use of `mask_state::UNALLOCATED`?

Reply:
Sorry, I'm not sure what you are asking. 
Calling `set_null_count(0)` here fixes the null-count value invalidated by the call to `mutable_view()` on `offsets_column`.

Reply:
The column is initially created with an unallocated mask. Does creating a mutable view invalidate the null count and allocate a mask? Or is it undefined behavior to write to a mutable view’s null mask if it is unallocated? (I'd have to look back at the implementation.)

Reply:
Yes, I suspect this will all be fixed once we get rid of `UNKNOWN_NULL_COUNT`.
https://github.com/rapidsai/cudf/blob/8a1ff584e39084b588b82a4d8a5f909bd7451916/cpp/src/column/column.cu#L130